plugins/nanospell/plugin.js

```diff
@@ -225,6 +225,18 @@
 				},
 				editorFocus: true
 			});
+			editor.addCommand('nanospellReset', {
+				exec: function(editor) {
+					spellcache = [];
+					suggestionscache = [];
+					ignorecache = [];
```

Comment:
something I just noticed, we instantiate these as arrays, then we use them like objects.  not sure if we actually benefit from this quirk